src/Web/WebPush.hs

```diff
@@ -225,6 +227,28 @@ mkPushNotification endpoint p256dh auth =
        , _pnMessage = ()
     }
 
+
+-- |Standard payload for web-pushes (but can be customized by passing
+-- any data to `sendPushNotification`, which has ToJSON instance)
+data PushNotificationMessage = PushNotificationMessage
+    { title :: T.Text
+    , body :: T.Text
+    , icon :: T.Text
+    , url :: T.Text
+    , tag :: T.Text
+    } deriving (Eq, Show)
+
+
+instance A.ToJSON PushNotificationMessage where
```

Comment:
Manual implementation has an advantage of little bit faster compilation, although it is not type safe, if we add new field, we will not be warned by compiler about missing it in ToJSON instance :( Can we use derived from Generic?